mods/old-this-pc-commands.wh.cpp

```diff
@@ -28,40 +28,33 @@ to `ExplorerFrame.dll` takes place to remove the redirect.
 
 #include <initguid.h>
 #include <windhawk_utils.h>
-#include <shlobj.h>
 #include <shobjidl.h>
-#include <ocidl.h>
 
-DEFINE_GUID(IID_IUnknown,         0x00000000, 0x0000, 0x0000, 0xc0,0x00, 0x00,0x00,0x00,0x00,0x00,0x46);
-DEFINE_GUID(IID_OpenControlPanel, 0xD11AD862, 0x66DE, 0x4DF4, 0xBF,0x6C, 0x1F,0x56,0x21,0x99,0x6A,0xF1);
-DEFINE_GUID(IID_IServiceProvider, 0x6D5140C1, 0x7436, 0x11CE, 0x80,0x34, 0x00,0xAA,0x00,0x60,0x09,0xFA);
-DEFINE_GUID(SID_STopLevelBrowser, 0x4c96be40, 0x915C, 0x11CF, 0x99,0xD3, 0x00,0xAA,0x00,0x4A,0xE8,0x37);
-
-HRESULT (*COpenControlPanel_Open)(void *pThis, LPCWSTR lpPage, LPCWSTR lpSubPage, IUnknown *pu);
-
-HRESULT OpenCplPage(IUnknown *pu, LPCWSTR lpPage, LPCWSTR lpSubPage)
+HRESULT OpenCplPage(IUnknown *punk, LPCWSTR lpPage, LPCWSTR lpSubPage)
 {
     IObjectWithSite *pows;
-    HRESULT hr = pu->QueryInterface(IID_IObjectWithSite, (void **)&pows);
+    HRESULT hr = punk->QueryInterface(IID_IObjectWithSite, (void **)&pows);
     if (SUCCEEDED(hr))
     {
-        IUnknown *pu2;
-        hr = pows->GetSite(IID_IUnknown, (void **)&pu2);
+        IUnknown *punk2;
+        hr = pows->GetSite(IID_PPV_ARGS(&punk2));
         if (SUCCEEDED(hr))
         {
-            void *pocp;
+            IOpenControlPanel *pocp = nullptr;
             hr = CoCreateInstance(
                 CLSID_OpenControlPanel,
                 NULL,
                 CLSCTX_INPROC_SERVER | CLSCTX_INPROC_HANDLER | CLSCTX_LOCAL_SERVER | CLSCTX_REMOTE_SERVER,
-                IID_OpenControlPanel,
-                &pocp
+                IID_PPV_ARGS(&pocp)
             );
             if (SUCCEEDED(hr))
             {
-                hr = COpenControlPanel_Open(pocp, lpPage, lpSubPage, pu2);
+                hr = pocp->Open(lpPage, lpSubPage, punk2);
+                pocp->Release();
```

Comment:
Perhaps you know that, but you can use [`winrt::com_ptr`](https://github.com/search?q=repo%3Aramensoftware%2Fwindhawk-mods+winrt%3A%3Acom_ptr&type=code) to have COM pointers released automatically for you.